influx-writer/influx-writer.py

```diff
@@ -86,6 +86,60 @@ def callback(self, ch, method, properties, body):
         return ret
 
 
+    def insert_to_listenstore(self, data, retries=5):
+        """
+        Inserts a batch of listens to the ListenStore. If this fails, then breaks the data into
+        two parts and recursively tries to insert them, until we find the culprit listen
+
+        Args:
+            data: the data to be inserted into the ListenStore
+            retries: the number of retries to make before deciding that we've failed
+
+        Returns: number of listens successfully sent
+        """
+
+        if not data:
+            return 0
+
+        done = False
+        failure_count = 0
+        while True:
+            try:
+                self.ls.insert(data)
+                done = True
```

Comment:
You could skip the whole done flag if you just returned len(data) here -- that would also simplify the code below.